### ISSUE TITLE
isLoading is always true if I poll query

### ISSUE THREAD

**Issue by aleksandrlat** [relay-tools/relay-hooks#159]
If I poll query like this
```
const { isLoading } = useQuery(query, vars, { networkCacheConfig: { poll: 5 * 60 * 1000 } })
```

`isLoading` is always true because Observable never completes when we use polling https://github.com/facebook/relay/blob/master/packages/relay-runtime/network/RelayObservable.js#L390

Should `isLoading` be set to `false` on next event for queries instead of complete?
https://github.com/relay-tools/relay-hooks/blob/master/src/FetchResolver.ts#L171-L176

**Comment by morrys** [comment]
Hi @aleksandrlat,
the fact that loading is managed in complete / error is precisely to identify when the network releases the request (e.g. when defer / stream will be officially released)

In this case I advise you to manage isLoading like this:

```ts
const {data, isLoading: isLoadingQuery} = useQuery(query, vars, {networkCacheConfig: {poll: 5 * 60 * 1000}});
const isLoading = isLoadingQuery && !data; 
```

**Comment by aleksandrlat** [comment]
I see. Thank you @morrys!

Is there a way to separate logic for defer / stream queries from regular queries?

**Comment by morrys** [comment]
It would be to define what is meant by regular queries, for me it is the queries that will be completed..

**Comment by aleksandrlat** [comment]
I guess for defer / stream queries we need to use some special directive in the query? Do I understand it correctly?

But if we poll regular query (query without defer / stream directive?) it never completes and `next` event indicates that network request completed. 

Or maybe we can have separate logic for poll queries?

**Comment by morrys** [comment]
> Or maybe we can have separate logic for poll queries?

@aleksandrlat, I am evaluating this PR https://github.com/relay-tools/relay-hooks/pull/160

**Comment by morrys** [comment]
fixed in relay-hooks [4.1.1](https://github.com/relay-tools/relay-hooks/releases/tag/v4.1.1)
